pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/DefaultRebalancePreChecker.java

```diff
@@ -105,37 +102,87 @@ private Boolean checkReloadNeededOnServers(String rebalanceJobId, String tableNa
       LOGGER.warn("Caught exception while trying to fetch reload status from servers", e);
     }
 
-    return needsReload;
+    return needsReload == null
+        ? RebalancePreCheckerResult.error("Could not determine needReload status, run manually")
+        : !needsReload ? RebalancePreCheckerResult.pass("No need to reload")
+            : RebalancePreCheckerResult.warning("Reload needed prior to running rebalance");
   }
 
   /**
    * Checks if minimize data movement is set for the given table in the TableConfig
    */
-  private boolean checkIsMinimizeDataMovement(String rebalanceJobId, String tableNameWithType,
+  private RebalancePreCheckerResult checkIsMinimizeDataMovement(String rebalanceJobId, String tableNameWithType,
       TableConfig tableConfig) {
     LOGGER.info("Checking whether minimizeDataMovement is set for table: {} with rebalanceJobId: {}", tableNameWithType,
         rebalanceJobId);
     try {
       if (tableConfig.getTableType() == TableType.OFFLINE) {
-        InstanceAssignmentConfig instanceAssignmentConfig =
-            InstanceAssignmentConfigUtils.getInstanceAssignmentConfig(tableConfig, InstancePartitionsType.OFFLINE);
-        return instanceAssignmentConfig.isMinimizeDataMovement();
+        boolean isInstanceAssignmentAllowed = InstanceAssignmentConfigUtils.allowInstanceAssignment(tableConfig,
+            InstancePartitionsType.OFFLINE);
+        InstanceAssignmentConfig instanceAssignmentConfig;
```

Comment:
this var declaration can be inlined with L125? as it's not used outside the if-block

Reply:
done